src/osqp_api.c

```diff
@@ -855,6 +861,15 @@ OSQPInt osqp_cleanup(OSQPSolver* solver) {
     if (work->timer) OSQPTimer_free(work->timer);
 # endif /* ifdef OSQP_ENABLE_PROFILING */
 
+# ifdef OSQP_ENABLE_DERIVATIVES
+      if (work->derivative_data){
+          if (work->derivative_data->y_l) OSQPVectorf_free(work->derivative_data->y_l);
+          if (work->derivative_data->y_u) OSQPVectorf_free(work->derivative_data->y_u);
+          if (work->derivative_data->rhs) OSQPVectorf_free(work->derivative_data->rhs);
+          c_free(work->derivative_data);
```

Comment:
It doesn't look like this is freeing all the vectors that are contained in the `derivative_data` struct. Please make sure to free all data.

Reply:
Yes - it looks like I don't free up `ryl` and `ryu`. I'll do this.

Reply:
Done